init-tools.cmd

```diff
@@ -0,0 +1,79 @@
+@if "%_echo%" neq "on" echo off
+setlocal
+
+:CheckOS
+IF "%PROCESSOR_ARCHITECTURE%"=="x86" (set bit=x86) else (set bit=x64)
+
+set INIT_TOOLS_LOG=%~dp0init-tools.log
+if [%PACKAGES_DIR%]==[] set PACKAGES_DIR=%~dp0packages\
+if [%TOOLRUNTIME_DIR%]==[] set TOOLRUNTIME_DIR=%~dp0build_tools
+set DOTNET_PATH=%~dp0.dotnet_stage0\Windows\%bit%\
+if [%DOTNET_CMD%]==[] set DOTNET_CMD=%DOTNET_PATH%dotnet.exe
+if [%BUILDTOOLS_SOURCE%]==[] set BUILDTOOLS_SOURCE=https://dotnet.myget.org/F/dotnet-buildtools/api/v3/index.json
+set /P BUILDTOOLS_VERSION=< "%~dp0BuildToolsVersion.txt"
+set BUILD_TOOLS_PATH=%PACKAGES_DIR%Microsoft.DotNet.BuildTools\%BUILDTOOLS_VERSION%\lib\
+set PROJECT_JSON_PATH=%TOOLRUNTIME_DIR%\%BUILDTOOLS_VERSION%
+set PROJECT_JSON_FILE=%PROJECT_JSON_PATH%\project.json
+set PROJECT_JSON_CONTENTS={ "dependencies": { "Microsoft.DotNet.BuildTools": "%BUILDTOOLS_VERSION%" }, "frameworks": { "netcoreapp1.0": { } } }
+set BUILD_TOOLS_SEMAPHORE=%PROJECT_JSON_PATH%\init-tools.completed
+set DOTNET_INSTALL_DIR=%~dp0.dotnet_stage0
+
+if DEFINED "%CHANNEL%" (
```

Comment:
The reason the PS1 script expected CHANNEL to be an env var, is because that PS1 script actually set the CHANNEL env var.  See [https://github.com/dotnet/cli/blob/rel/1.0.0/build_projects/dotnet-cli-build/build.ps1#L54-L60](https://github.com/dotnet/cli/blob/rel/1.0.0/build_projects/dotnet-cli-build/build.ps1#L54-L60).  This was setting CHANNEL to the value set in the branchinfo.txt file.

---
In reply to: [68847256](https://github.com/dotnet/cli/pull/3719#discussion_r68847256) [](ancestors = 68847256)